src/dmd/aggregate.d

```diff
@@ -372,16 +370,29 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
                 if (!MODimplicitConv(v2.type.mod, vd.type.mod))
                     vd.overlapUnsafe = true;
 
-                if (!vx)
+                if (i > j)
                     continue;
-                if (v2._init && v2._init.isVoidInitializer())
+
+                if (!v2._init)
                     continue;
 
-                if (vx._init && v2._init)
+                if (v2._init.isVoidInitializer())
+                    continue;
+
+                if (vd._init && !vdIsVoidInit && v2._init)
                 {
                     .error(loc, "overlapping default initialization for field `%s` and `%s`", v2.toChars(), vd.toChars());
                     errors = true;
                 }
+                else if (v2._init && i < j)
+                {
+                    // @@@DEPRECATED_v2.091@@@.
+                    // Made an error in 2.086.
```

Comment:
This isn't an error; it's a deprecation warning.  Do the comments need updating, or does the code need to be made an error?

Reply:
I thinks the numbers just need to be swapped.

Reply:
Is it ok now? At the time of the PR I took example from something in typesem.d but I see that now it has been modified.